src/main/scala/is/hail/expr/ir/PointedTypeInfo.scala

```diff
@@ -0,0 +1,32 @@
+package is.hail.expr.ir
+
+import is.hail.asm4s._
+
+sealed trait PointedTypeInfo[T] {
```

Comment:
This seems unnecessary.  Java types are already pointed: these is a default initialization value (0 or null) which can always be generated with a single bytecode instruction.  Also, there is a bewildering proliferation of TypeInfo-like classes that I'm starting to be suspicious of.

Reply:
I need to access the default value while only knowing `TypeInfo[_]`. I don't know how to do that, since I don't actually have the scala type.

Reply:
Seem my other comment.  You should be able to get the `Type` for an IR.